actix-limitation/src/lib.rs

```diff
@@ -70,16 +77,27 @@ pub const DEFAULT_PERIOD_SECS: u64 = 3600;
 pub const DEFAULT_COOKIE_NAME: &str = "sid";
 
 /// Default session key.
+#[cfg(feature = "session")]
 pub const DEFAULT_SESSION_KEY: &str = "rate-api-id";
 
+/// Helper trait to impl Debug on GetKeyFn type
+pub trait GetKeyFnT: Fn(&ServiceRequest) -> Option<String> + Sync {}
+impl<T> GetKeyFnT for T where T: Fn(&ServiceRequest) -> Option<String> + Sync {}
+/// Get key resolver function type
+pub type GetKeyFn = Box<dyn GetKeyFnT + Send + Sync>;
+impl fmt::Debug for GetKeyFn {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "GetKeyFn")
+    }
+}
+
 /// Rate limiter.
-#[derive(Debug, Clone)]
+#[derive(Debug)]
```

Comment:
This is the only reason I've seen to mark the PR breaking, if GetKeyFn could be Rc instead of Box it could be semver-minor

Reply:
Oh ! I forgot about this, I wrapped the Closure field in an Arc, 
```rs
    get_key_fn: Arc<GetKeyFn>,
```
Is this what you wanted me to do ?

Reply:
:sweat: Rc doesn't work because it's not Send
https://github.com/actix/actix-extras/runs/8071465240?check_suite_focus=true#step:6:45

I reverted back to the original Arc instead.